scripts/g.extension/g.extension.py

```diff
@@ -2299,46 +2317,54 @@ def resolve_source_code(url=None, name=None, branch=None, fork=False):
     >>> resolve_source_code('https://bitbucket.org/joe-user/grass-module') # doctest: +SKIP
     ('remote_zip', 'https://bitbucket.org/joe-user/grass-module/get/default.zip')
     """
+    # Handle URL for the offical repo
     if not url and name:
         module_class = get_module_class_name(name)
-        # note: 'trunk' is required to make URL usable for 'svn export' call
+        repo_branches = get_branches(
+            "https://api.github.com/repos/OSGeo/grass-addons/branches", True
+        )
+        # Define branch to fetch from (latest or current version)
+        version_branch = "grass{}".format(version[0])
+        if version_branch not in repo_branches:
+            version_branch = repo_branches[-1]
+
+        # Set URL for the given GRASS version
         git_url = (
-            "https://github.com/OSGeo/grass-addons/trunk/"
-            "grass{version}/{module_class}/{module_name}".format(
-                version=version[0], module_class=module_class, module_name=name
-            )
+            "https://github.com/OSGeo/grass-addons/branches/"
+            f"{version_branch}/src/{module_class}/{name}"
         )
-        # trac_url = 'https://trac.osgeo.org/grass/browser/grass-addons/' \
-        #            'grass{version}/{module_class}/{module_name}?format=zip' \
-        #            .format(version=version[0],
-        #                    module_class=module_class, module_name=name)
-        # return 'official', trac_url
         return "official", git_url
 
+    # Handle URL for a fork of the offical repo
     if url and fork:
         module_class = get_module_class_name(name)
 
         # note: 'trunk' is required to make URL usable for 'svn export' call
-        if branch in ["master", "main"]:
+        if branch is None:
+            repo_branches = get_branches(
+                url.rstrip("/").replace("github.com/", "api.github.com/repos/")
+                + "/branches",
+                True,
+            )
+            # Define branch to fetch from (latest or current version)
+            if not repo_branches:
+                svn_reference = "trunk"
+            else:
+                version_branch = "grass{}".format(version[0])
+                if version_branch not in repo_branches:
+                    version_branch = repo_branches[-1]
+                svn_reference = "branches/{}".format(version_branch)
+        elif branch in ["master", "main"]:
             svn_reference = "trunk"
```

Comment:
master and main are really just names for branches which are branches like any other branch. As far as I understand, in this GitHub Git-Subversion bridge, trunk refers to the default branch. It is not a nickname for master or main, so there is no reason to translate master and main to trunk. What needs to resolve to trunk is the state when no branch is provided, i.e., you are asking for the default branch which is what trunk refers to.

https://docs.github.com/en/github/importing-your-projects-to-github/working-with-subversion-on-github/support-for-subversion-clients
https://docs.github.com/en/github/administering-a-repository/managing-branches-in-your-repository/changing-the-default-branch

Reply:
Yes. If we get branches, we can also get the default branch. For github, it could be done like this:

```
def get_github_default_branch(organization="OSGeo", repository="grass-addons"):
    """Get default branch for github repo using github API"""
    req = urlrequest.urlopen(f"https://api.github.com/orgs/{organization}/repos")
    content = json.loads(req.read())
    default_branch = [repo["default_branch"] for repo in content if repo["name"] == repository][0]
    return default_branch
``` 
That function could be included / merged with get_branches function.

But for the official repo, that is probably not necessary (as we use the version branches) and as you say, `trunk` refers to the default branch anyway, so as long as we use svn export that should be OK. For all other repos user specific repos, users can specify a branch. But there we would need a solution for bitbucket, gitlab ... too...

Reply:
I agree that it is not needed for the official repo. For other repos, I think it would be fine to get the default branch for GitHub and for others, at least for now, either require branch to be set or use master or main as a (internally set) default value. I think that is still reasonable behavior which can documented clearly enough. However, I'm fine with requiring users to set branch explicitly too, but not having to do that is closer to Git and pip behavior.

`get_github_default_branch` looks like a separate function from _get branches_.